invenio_app_rdm/records_ui/views/decorators.py

```diff
@@ -381,3 +382,24 @@ def view(*args, **kwargs):
         return response
 
     return view
+
+
+def secret_link_or_login_required():
+    """Skip login redirection check for requests with secret links.
+
+    If access has been granted via a secret link, then permissions are checked
+    in the dedicated view.
+    """
+
+    def decorator(f):
+        @wraps(f)
+        def view(**kwargs):
+            secret_link_token_arg = "token"
+            session_token = session.get(secret_link_token_arg, None)
```

Comment:
one clarification, I think we should leave a comment on this in the code on how the token is already in the session, since more "obvious" to understand how this works is to check if token is in the request args

general thoughts: it was not clear to me how at this stage the token is already in the session, so I started digging and maybe it makes sense that it was already loaded there by identity change signal (are we sure the signal will be processed before we enter this code block? I guess we can assume the identity loader always goes first, otherwise sometimes shared links would fail - not sure if this is possible)

an alternative would be checking if token is there as a request argument, like it is done in verify_token function, but it might be redundant

Reply:
Token is not present here as a request argument, as the request at this point is created by clicking on "Edit" or "New version" buttons, which is a simple create draft request (not the secret link that was copied). We could retrieve token from the request's `Referer` header, but it is not a reliable source of information at this point IMHO.
And yes, you are right, we can assume that the `on_identity_loaded` will be called before the deposit edit view